java/pattern/variable/UnboundValueVariable.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.typeql.lang.pattern.variable;
+
+import com.vaticle.typeql.lang.pattern.constraint.Constraint;
+import com.vaticle.typeql.lang.pattern.constraint.ValueConstraint;
+import com.vaticle.typeql.lang.pattern.variable.builder.ValueVariableBuilder;
+
+import java.util.Collections;
+import java.util.List;
+
+public class UnboundValueVariable extends UnboundVariable implements ValueVariableBuilder {
+    UnboundValueVariable(Reference reference) {
```

Comment:
would this look better if we had `Reference.Value` as a subtype, and `Reference.Concept` for the rest? Is there much benefit?